README.md

```diff
@@ -24,6 +24,64 @@ Library for accessing [libkiwix](https://github.com/kiwix/libkiwix) and [libzim]
 
 AAR file will be generated in directory `lib/build/outputs/aar`
 
+### Load zim file
+
+To load a ZIM file you need to create an `Archive` object.
+
+```kotlin
+val archive = Archive("your-file-path")
+```
+
+### Load main page
+
+The `mainPage` property is used to retrieve the path of the main entry page for a Kiwix content archive.
+If the main entry is a redirect, it will fetch the path of the redirected item;
+otherwise, it will return the path of the main entry itself.
+If the main entry is not found, the archive will throw an `EntryNotFoundException`.
+
+```kotlin
+val mainPage: String?
+    get() =
+        try {
+            archive.mainEntry.getItem(true).path
+        } catch (exception: Exception) {
```

Comment:
@mgautierfr, Thanks, one more thing we have some other methods e.g. `getEntryByPath`, and `getEntryByTitle` which can also throw `EntryNotFoundException`, but if I catch `EntryNotFoundException` it crashed the process but the error message is same `Caused by: java.lang.Exception: Cannot find entry` so it should throw `EntryNotFoundException` but it is throwing `java.lang.Exception`, that's why I have caught an exception this way to avoid unnecessary crashes.

Reply:
This sound like a bug.
`getEntryByPath` should throw a `EntryNotFoundException`, not another exception.
And this is tested here : https://github.com/kiwix/java-libkiwix/blob/main/lib/src/test/test.java#L158-L173

Reply:
@mgautierfr, You are right `getEntryByPath` is throwing `EntryNotFoundException`, but we are using the `Archive` class with webview in `kiwix-android`. 

The problem is `getEntryByPath` method is throwing an `EntryNotFoundException` exception that is being caught by the WebView's `shouldInterceptRequest` method, and then it's causing an unhandled `InvocationTargetException`. and this exception is being caught as a generic Exception [more info](https://www.javatpoint.com/invocationtargetexception). This is the reason it is throwing `java.lang.Exception`.

So here currently we are compiling `java-libkiwix` for Android and the `README.md` should show how to load ZIM files and articles in the Android application in Android we are using these methods with webview so we should show how to properly use these functions, without any unnecessary crash.

logs:

```
Client raised exception in shouldInterceptRequest. Re-throwing on UI thread.
The following exception was raised by shouldInterceptRequest:
FATAL EXCEPTION: main
Process: org.kiwix.kiwixmobile, PID: 16410
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:586)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1073)
Caused by: java.lang.reflect.InvocationTargetException
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:576)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1073) 

Caused by: java.lang.Exception: Cannot find entry
at org.kiwix.libzim.Archive.getEntryByPath(Native Method)
at org.kiwix.kiwixmobile.core.reader.ZimFileReader.getActualUrl(ZimFileReader.kt:184)
at org.kiwix.kiwixmobile.core.reader.ZimFileReader.getActualUrl$default(ZimFileReader.kt:181)
at org.kiwix.kiwixmobile.core.reader.ZimFileReader.getItem(ZimFileReader.kt:264)
at org.kiwix.kiwixmobile.core.reader.ZimFileReader.getMimeTypeFromUrl(ZimFileReader.kt:165)
at org.kiwix.kiwixmobile.core.reader.ZimReaderContainer.load(ZimReaderContainer.kt:53)
at org.kiwix.kiwixmobile.core.main.CoreWebViewClient.shouldInterceptRequest(CoreWebViewClient.kt:150)
at WV.ga.a(chromium-TrichromeWebViewGoogle.aab-stable-579013931:79)
at org.chromium.android_webview.AwContentsBackgroundThreadClient.shouldInterceptRequestFromNative(chromium-TrichromeWebViewGoogle.aab-stable-579013931:15)
```

Reply:
> The problem is getEntryByPath method is throwing an EntryNotFoundException exception that is being caught by the WebView's shouldInterceptRequest method, and then it's causing an unhandled InvocationTargetException. and this exception is being caught as a generic Exception [more info](https://www.javatpoint.com/invocationtargetexception). This is the reason it is throwing java.lang.Exception.

As you can see in the documentation you've pointed, you can get the original exception using `getCause()`.
You should use it to handle correctly the exception. The best is probably to re throw it. 
See https://stackoverflow.com/questions/10214525/re-throw-an-invocationtargetexception-target-exception

> So here currently we are compiling java-libkiwix for Android and it would be nice to show in README.md how to load ZIM files and articles in the Android application, in Android we are using these methods with webview. so we should show, how to properly use these functions, without any unnecessary crashes.

No. In **kiwix-** android you are using a webview. What your are describing is not the use of `Archive::getEntryByPath`. This is the use of a method called `getEntryByPath` of a webview (and it is not the same).

Here you are describing how to use `java-libkiwix`, so you must stick to the API. If you add another layer in other project, it is your problem, not the `java-libkiwix` one.